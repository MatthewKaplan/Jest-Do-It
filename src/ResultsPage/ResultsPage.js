import React from "react";
import '../Styles/_WelcomePage.scss';

const ResultsPage = props => {
  let displayBtn = props.secondRound === false ? <input className="playAgain" type='submit' data-test='wrong-answers-btn' onClick={props.switchSecondRound} value='Practice the ones you got wrong' /> : null;
  return (
    <section className="welcomePage">
      <article className="backGround">
        <div className="welcomeContent">
          <span className="welcomeHeader">{props.secondRound === false ? props.correctQuestions.length >= 23 ? 'Good Job!' : 'You can do better!' : 'Thats Better!'}</span>
          <p className="welcomeHeader"><span>{props.isLoading ? props.isLoadingFunction : `${Math.round(props.correctQuestions.length * 100 / props.questions.length)}%`}</span></p>
          <input className="playAgain" 
              type='submit'
              onClick={props.playAgain}
              data-test='play-again-btn'
              value='Play Again!'
          /><br/><br/><br/>
          {displayBtn}
        </div>
      </article>
    </section>
  );
};

export default ResultsPage;