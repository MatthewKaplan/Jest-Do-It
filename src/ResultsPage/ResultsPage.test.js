import React from 'react';
import ResultsPage from './ResultsPage';
import { shallow } from 'enzyme';

const mockCorrectQuestions =
{
id: 1,
question: "Jest is a JavaScript testing tool originally created by the developers at:",
potentialAnswers: [
  "Facebook",
  "Twitter",
  "Airbnb",
  "Microsoft"
],
correctAnswer: "Facebook",
save: "false",
onCorrectAnswer: "Thats correct! The developers over at Facebook who are also responsable for creating React brought us Jest.",
onIncorrectAnswer: "Thats incorrect. The correct answer is: Facebook.",
linkName: "Jest Github",
link: "https://github.com/facebook/jest"
}

const mockQuestions = 
{
id: 2,
question: "What tool can you use in combination with Jest?",
potentialAnswers: [
  "Enzyme",
  "Mocha",
  "Jtest",
  "Cucumber"
],
correctAnswer: "Enzyme",
save: "false",
onCorrectAnswer: "Thats correct! Although you can use other tools with Jest such as Chai, Enzyme is a great tool to use in comination with Jest.",
onIncorrectAnswer: "Thats incorrect. The correct answer is: Enzyme.",
linkName: "Enzyme Docs",
link: "https://airbnb.io/enzyme/"
}

const mock_playAgain = jest.fn()
const mock_switchSecondRound = jest.fn()

describe("ResultsPage", () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ResultsPage
        correctQuestions={mockCorrectQuestions}
        questions={mockQuestions} 
        playAgain={mock_playAgain}
        switchSecondRound={mock_switchSecondRound}
     />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should invoke the handleChangeCard method onClick", () => {
    wrapper.find("[data-test='play-again-btn']").simulate('click', { preventDefault: () => {} });
    expect(mock_playAgain).toBeCalled();
  });

  //  it("should invoke the switchSecondRound method onClick", () => {
  //   wrapper.find("[data-test='wrong-answers-btn']").simulate('click', { preventDefault: () => {} });
  //   expect(mock_switchSecondRound).toBeCalled();
  // });
});













