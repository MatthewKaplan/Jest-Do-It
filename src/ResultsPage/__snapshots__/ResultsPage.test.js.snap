// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPage should match snapshot 1`] = `
<section
  className="welcomePage"
>
  <article
    className="backGround"
  >
    <div
      className="welcomeContent"
    >
      <span
        className="welcomeHeader"
      >
        Thats Better!
      </span>
      <p
        className="welcomeHeader"
      >
        <span>
          NaN%
        </span>
      </p>
      <input
        className="playAgain"
        data-test="play-again-btn"
        onClick={[MockFunction]}
        type="submit"
        value="Play Again!"
      />
      <br />
      <br />
      <br />
    </div>
  </article>
</section>
`;
